### PR TITLE
✨ Added endpoint to get basic offender detail -

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**
included the probation status

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java [hunk 1]
```diff
@@ -119,4 +119,7 @@ private ProbationRecord addConvictionsToProbationRecord(ProbationRecord probatio
         return defaultClient.getConvictionRequirements(crn, convictionId).block();
     }
 
+    public OffenderDetail getOffenderDetail(String crn) {
+        return defaultClient.getOffenderDetailByCrn(crn).block();
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java:123]: Could return a Mono all the way to the controller here which would keep it fully non-blocking. Doesn't really matter that much in the whole scheme of things so up to you

File: src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/OffenderRestClientIntTest.java [hunk 2]
```diff
@@ -140,4 +141,30 @@ public void givenOffenderDoesNotExist_whenGetOffenderMatchDetail_thenReturnNull(
...
+        var optionalOffender = offenderRestClient.getOffenderDetailByCrn(CRN).blockOptional();
+
+        assertThat(optionalOffender).isNotEmpty();
+        var offenderDetail = optionalOffender.get();
+        assertThat(offenderDetail.getTitle()).isEqualTo("Mr.");
+        assertThat(offenderDetail.getProbationStatus()).isSameAs(ProbationStatus.CURRENT);
+        assertThat(offenderDetail.getDateOfBirth()).isEqualTo(LocalDate.of(2000, Month.JULY, 19));
+        assertThat(offenderDetail.getForename()).isEqualTo("Aadland");
+        assertThat(offenderDetail.getSurname()).isEqualTo("Bertrand");
+        assertThat(offenderDetail.getTitle()).isEqualTo("Mr.");
```
Comment (jhevans) [inline src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/OffenderRestClientIntTest.java:156]: Duplicate assertion, see line 151

File: src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/communityapi/mapper/OffenderMapperTest.java [hunk 3]
```diff
@@ -101,6 +101,26 @@ void shouldMapOffenderDetailsToOffender() {
...
+
+        assertThat(offenderDetail.getOtherIds().getCrn())
+            .isNotNull()
+            .isEqualTo("X320741");
+        assertThat(offenderDetail.getTitle()).isEqualTo("Mr.");
+        assertThat(offenderDetail.getOtherIds().getCrn()).isEqualTo("X320741");
+        assertThat(offenderDetail.getMiddleNames()).containsExactlyInAnyOrder("Hope", "Felix");
+        assertThat(offenderDetail.getProbationStatus()).isSameAs(ProbationStatus.CURRENT);
+        assertThat(offenderDetail.getDateOfBirth()).isNull();
+        assertThat(offenderDetail.getTitle()).isEqualTo("Mr.");
```
Comment (jhevans) [inline src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/communityapi/mapper/OffenderMapperTest.java:119]: Duplicate assertion, see  line 114

**Review by jhevans — APPROVED** [review]
A few very minor things

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java (hunk 1 above)
Comment (chris-faulkner-mojdt) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/OffenderService.java]: Done

File: src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/OffenderRestClientIntTest.java (hunk 2 above)
Comment (chris-faulkner-mojdt) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/OffenderRestClientIntTest.java]: Done

File: src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/communityapi/mapper/OffenderMapperTest.java (hunk 3 above)
Comment (chris-faulkner-mojdt) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/restclient/communityapi/mapper/OffenderMapperTest.java]: Done

**Review by jhevans — APPROVED** [review]
👍

**Pull request merged**